src/introduction/hal-design-philosophy.md

```diff
@@ -1 +1,17 @@
 # HAL Design Philosophy
+
+The HAL was designed based on some recommended patterns and rules that are described in several documents:
+- [`esp-hal` developers' guidelines][guidelines]
+- [The Embedded Rust Book - HAL Design Patterns][embedded-rust-patterns]
+
+[guidelines]: https://github.com/esp-rs/esp-hal/blob/main/documentation/DEVELOPER-GUIDELINES.md
+[embedded-rust-patterns]: https://docs.rust-embedded.org/book/design-patterns/hal/index.html
+
+The HAL implements both blocking and asynchronous APIs for the supported peripherals. When applicable, drivers implement the [`embedded-hal`][embedded-hal] and [`embedded-hal-async`][embedded-hal-async] traits.
+
+## `Peripheral` Pattern
+
+`esp-hal` drivers take pins and peripherals as `peripheral::Peripheral` in most circumstances. This means you can pass the pin/peripheral or a mutable reference to the pin/peripheral.
```

Comment:
Wanted to add a link to https://docs.espressif.com/projects/rust/esp-hal/1.0.0-beta.0/esp32c6/esp_hal/index.html#peripheral-pattern but https://docs.espressif.com/projects/rust/esp-hal/latest/esp32c6/esp_hal/index.html#peripheral-pattern doesn't work

Reply:
Probably there is a way (with some JavaScript?) to make it work with the generated `latest/index.html` thing - in this case we could just duplicate the two sentences since they probably won't change and it not much - but in general would be certainly good if that would work

Reply:
> Wanted to add a link to https://docs.espressif.com/projects/rust/esp-hal/1.0.0-beta.0/esp32c6/esp_hal/index.html#peripheral-pattern but https://docs.espressif.com/projects/rust/esp-hal/latest/esp32c6/esp_hal/index.html#peripheral-pattern doesn't work

We never re-deployed the documentation after the related changes so this shouldn't be expected to work.